### PR TITLE
[12.0][MIG] crm_claim_type: Migration to 12.0

### PR DESCRIPTION
**Pull request by Tardo**
PR based on: https://github.com/OCA/crm/pull/193
Rebased some commits

Cc @tecnativa

### CODE REVIEW

**Review by carlosdauden — APPROVED** [review]

File: crm_claim_type/views/crm_claim_stage.xml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                <newline/>
+                <field name="claim_common" />
+                <field name="claim_type"
+                     attrs="{'required':[('claim_common','=',False)],
+                     'invisible':[('claim_common','=',True)]}"/>
+            </xpath>
+        </field>
+    </record>
+
+    <record id="crm_stage_claim_type_tree" model="ir.ui.view">
```
Comment (tbaden) [inline crm_claim_type/views/crm_claim_stage.xml:25]: optional:
use original id (crm_claim_stage_tree)

File: crm_claim_type/views/crm_claim_stage.xml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+
+    <record id="crm_claim.crm_claim_category_claim0" model="ir.actions.act_window">
+        <field name="context">
+            {"search_default_user_id":uid, "stage_type":'claim'}
+        </field>
+    </record>
+
+    <record id="crm_stage_claim_type_form" model="ir.ui.view">
```
Comment (tbaden) [inline crm_claim_type/views/crm_claim_stage.xml:10]: optional:
use original id (crm_claim_stage_form)

**Review by tbaden — APPROVED** [review]
thanks for the work 👍 
I did a code review: LGTM, if you like some optional changes.

**Review by chienandalu — APPROVED** [review]
Tested on runbot

**Pull request merged**